File: AnJeongHoo/Java/src/main/java/com/algorithm/jeongh00/back1697.java
import java.util.*;

class Main {

    static int visited[] = new int[100001];

    public static int BFS(int N, int K) {

        Queue<Integer> q = new LinkedList<Integer>();

        q.add(N);
        int index = N;
        int n = 0;
        visited[index] = 1;

        while (!q.isEmpty()) {

            n = q.poll();

            if (n == K) {
                return visited[n] - 1;
            }

            if (n - 1 > 0 && visited[n - 1] == 0) {
                visited[n - 1] = visited[n] + 1;
                q.add(n - 1);
            }

            if (n + 1 <= 100000 && visited[n + 1] == 0) {
                visited[n + 1] = visited[n] + 1;
                q.add(n + 1);
            }

            if (2 * n <= 100000 && visited[2 * n] == 0) {
                visited[2 * n] = visited[n] + 1;
                q.add(2 * n);
            }
        }
        return -1;
    }

    public static void main(String[] args) {

        Main T = new Main();
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int K = sc.nextInt();

        int result = T.BFS(N, K);
        System.out.println(result);
    }
}

File: JeongYunSung/Kotlin/src/main/kotlin/com/yunseong/algorithm/back1697/Back1697.kt
package com.yunseong.algorithm.back1697

import java.util.LinkedList

fun main(args: Array<String>) {
    val (n, k) = readln().split(" ").map { it.toInt() }

    val queue = LinkedList<Pair<Int, Int>>()
    val visited = BooleanArray(100001) { false }

    queue.add(n to 0)

    while (queue.isNotEmpty()) {
        val x = queue.poll()

        if (x.first == k) {
            println(x.second)
            break
        }

        if (x.first >= 0 && x.first < visited.size && !visited[x.first]) {
            visited[x.first] = true

            queue.add(x.first - 1 to x.second + 1)
            queue.add(x.first + 1 to x.second + 1)
            queue.add(x.first * 2 to x.second + 1)
        }
    }
}
